### PR TITLE
 022 Script for requirement  [PerformAudioPassThru] audioPassThruIcon does NOT exist at 'AppStorageFolder'

### PR DESCRIPTION
**Pull request by maria-mih**
022_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua
All common functions used available at #703
Testing coverage available at #643
@istoimenova Please review

### CODE REVIEW

**Comment by istoimenova** [comment]
@maria-mih, the script that is listed in TestSet #643 with ID - 035 has other name, please update the Test Set.

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+-- transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--respond with <resultCode_received_from_HMI> + <info> to mobile app (expected resultCode from HMI is "WARNINGS" and info: "Reference image(s) not found" 
+-- as described in [F-S] Conditions for SDL to handle case when requested <Image> does not exist at "AppStorageFolder"
+--
+--1. Used preconditions
+--1.1. audioPassThruIcon does not exist in created by app AppStorageFolder
+--1.2. Request is sent with valid audioPassThruIcon parameter with type DYNAMIC and with all mandatory parameters within bounds (samplingRate, maxDuration, bitsPerSample, audioType)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, mandatory params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, mandatory params) to HMI
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @maria-mih, points 2.1 / 2.2 are not steps, they are expectations of the step "Send..." and should be separate, something like:

1. Send PerformAudioPassThru (audioPassThruIcon, other params) from mobile to SDL and check:
SDL sends UI.PerformAudioPassThru
SDL sends TTS.Speak to HMI
2. HMI sends UI.PerformAudioPassThru (WARNINGS) to SDL

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_Check_audioPassThruIcon_Existence()
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @maria-mih, as purpose of script is to check requirement for not existing audioPassThruIcon, will be better to rename the name of precondition, as example:
Precondition_Check_audioPassThruIcon_NotExist

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @maria-mih, please move parameters on previous row.

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+--respond with <resultCode_received_from_HMI> + <info> to mobile app (expected resultCode from HMI is "WARNINGS" and info: "Reference image(s) not found" 
+-- as described in [F-S] Conditions for SDL to handle case when requested <Image> does not exist at "AppStorageFolder"
+--
+--1. Used preconditions
+--1.1. audioPassThruIcon does not exist in created by app AppStorageFolder
+--1.2. Request is sent with valid audioPassThruIcon parameter with type DYNAMIC and with all mandatory parameters within bounds (samplingRate, maxDuration, bitsPerSample, audioType)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, mandatory params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, mandatory params) to HMI
+--2.2 SDL sends TTS.Speak to HMI
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @maria-mih, as mandatory parameters are used only by UI, you should not expect TTS part.
Please update also line 26.

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:TestStep_Mandatory_Params_Missing_audioPassThruIcon_DYNAMIC()
+  local CorIdPerfAudioPassThruOnlyMandatory= self.mobileSession:SendRPC("PerformAudioPassThru",
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @maria-mih, you should be sure that RPC is allowed by policy.

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      maxDuration = 2000,
+      bitsPerSample = "8_BIT",
+      audioType = "PCM", 
+      audioPassThruIcon =
+      { value = "icon.png",
+        imageType = "DYNAMIC"
+      }      
+    })
+
+  -- hmi expects UI.PerformAudioPassThru request
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @maria-mih, please remove such comments.

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  EXPECT_HMICALL("UI.PerformAudioPassThru",
+    {
+      appID = self.applications[applicationName],
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @maria-mih, "applicationName" is not defined, you can use: config.application1.registerAppInterfaceParams.appName

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua (hunk 1 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @istoimenova Updated

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @istoimenova Updated

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @istoimenova Updated

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @istoimenova Done

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua (hunk 5 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @istoimenova Done

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua (hunk 6 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @istoimenova Removed

File: test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua (hunk 7 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/036_ATF_P_PerformAudioPassThru_WARNING_Missing_audioPassThruIcon_MandatoryParams_DYNAMIC.lua]: @istoimenova Updated

**Comment by maria-mih** [comment]
@istoimenova Please start second review iteration, updated script available on commit 8d5d306cd9c14ba18acafb13ee7ecaac213c06d5

**Pull request merged**